uadapy/plotting/plots2D.py

```diff
@@ -97,18 +97,36 @@ def plot_contour(distributions, resolution=128, ranges=None, quantiles:list=None
         distributions = [distributions]
     contour_colors = generate_spectrum_colors(len(distributions))
 
+    # Determine default quantiles: 25%, 75%, and 95%
+    if quantiles is None:
+        quantiles = [25, 75, 95]
+    largest_quantile = max(quantiles)
+
     if ranges is None:
         min_val = np.zeros(distributions[0].mean().shape)+1000
         max_val = np.zeros(distributions[0].mean().shape)-1000
         cov_max = np.zeros(distributions[0].mean().shape)
+
+        # Dynamically set the expansion factor based on the largest quantile
+        if largest_quantile >= 99.99:
```

Comment:
This only holds for normal distributions. Wouldn't it be more generalizable if you used the samples to estimate the ranges?